hudi-common/src/main/java/org/apache/hudi/common/config/HoodieStorageConfig.java

```diff
@@ -243,7 +243,12 @@ public class HoodieStorageConfig extends HoodieConfig {
       .withDocumentation("The fully-qualified class name of the factory class to return readers and writers of files used "
           + "by Hudi. The provided class should implement `org.apache.hudi.io.storage.HoodieIOFactory`.");
 
-
+  public static final ConfigProperty<String> HOODIE_STORAGE_CLASS = ConfigProperty
```

Comment:
I think we can replace `HOODIE_IO_FACTORY_CLASS` with `HOODIE_STORAGE_CLASS` since wherever the `HoodieIOFactory` is instantiated, the `HoodieStorage` is needed.  So we can add a new API `getIOFactory()` in `HoodieStorage` to get the `HoodieIOFactory` instance and only use `HOODIE_STORAGE_CLASS` if the instantiation through reflection is needed.

Reply:
makes sense